ui/libs/pages/search/src/lib/components/filter/filter.component.ts

```diff
@@ -0,0 +1,116 @@
+import {Component, EventEmitter, Input, Output, ViewChild} from '@angular/core';
+import { addFq, IFilter, removeFq } from '@eosc-search-service/search';
+import { ActivatedRoute, Router } from '@angular/router';
+import { FlatNode } from '@eosc-search-service/layout';
+import { FormControl } from '@angular/forms';
+import { debounceTime } from 'rxjs';
+import { UntilDestroy, untilDestroyed } from '@ngneat/until-destroy';
+import {assertNotNull} from "@eosc-search-service/types";
+
+@UntilDestroy()
+@Component({
+  selector: 'ess-filter',
+  template: `
+    <div class="filter" *ngIf="filter">
+      <span class="filter-title"
+        ><b>{{ filter.title }}</b></span
+      >
+      <input
+        *ngIf="filter.showMore"
+        [attr.placeholder]="'Search in ' + filter.title"
+        class="query-input form-control form-control-sm"
+        [formControl]="queryFc"
+      />
+      <div [ngClass]="{filter__viewport: filter.showMore}" (scroll)="onScroll($event)" #content>
+        <ess-checkboxes-tree
+          [data]="filter.data"
+          (checkboxesChange)="addFilter($event)"
+        ></ess-checkboxes-tree>
+      </div>
```

Comment:
Why haven't you used ready build-in CDK infinity scroll solution?

Reply:
infinite scroll requires`* cdkVirtualFor` which we do not use directly. I do assume it might be possible to make it work with `ng-zorro`'s checkboxes, but I was a little short on time with this one. A thing to improve, for sure. (Either cdk could be used, or at least shananigans with scroll could be extracted to separated directive)